### PR TITLE
Move package json inside webpack directory

### PR DESCRIPTION
**Pull request by justin808**
1. Changed everything client side to be under /client directory
2. Moved node_modules to /client directory
3. Updated libraries including switching to babel.

Addresses:
Move package.json to inside webpack directory #10

### CODE REVIEW

**Comment by mbreining** [comment]
Looks good.

**Comment by justin808** [comment]
This code is running on heroku right now.

File: client/assets/javascripts/CommentBox.jsx [hunk 1]
```diff
@@ -5,12 +5,12 @@ var React = require('react/addons');
...
-var Nav = require('react-bootstrap/Nav')
-var NavItem = require('react-bootstrap/NavItem')
+var Input = require('react-bootstrap/lib/Input');
+var Button = require('react-bootstrap/lib/Button');
+var Row = require('react-bootstrap/lib/Row');
+var Col = require('react-bootstrap/lib/Col');
+var Nav = require('react-bootstrap/lib/Nav')
+var NavItem = require('react-bootstrap/lib/NavItem')
 
 var marked = require("marked");
```
Comment (bdefore) [inline client/assets/javascripts/CommentBox.jsx]: Is this change intended? On another project, I have been using these paths without specifying lib.

File: client/webpack.rails.config.js [hunk 2]
```diff
@@ -7,18 +7,19 @@
...
 config.entry.push("./scripts/rails_only"); // rails specific assets
-config.output = { filename: "webpack-bundle.js",
+config.output = { filename: "client-bundle.js",
                   path: "../app/assets/javascripts" };
 config.externals = { jquery: "var jQuery" }; // load jQuery from cdn or rails asset pipeline
 config.module.loaders.push(
-  { test: /\.jsx$/, loaders: ["es6", "jsx?harmony"] },
-  // Next 2 lines expose jQuery and $ to any JavaScript files loaded after webpack-bundle.js
+  { test: /\.jsx$/, exclude: /node_modules/, loader: 'babel-loader'},
+  // Next 2 lines expose jQuery and $ to any JavaScript files loaded after client-bundle.js
```
Comment (bdefore) [inline client/webpack.rails.config.js]: I believe you might want some .jsx from node_modules to be included? For example, I use the react-paginate library.

File: client/assets/javascripts/CommentBox.jsx (hunk 1 above)
Comment (justin808) [inline client/assets/javascripts/CommentBox.jsx]: Yes, part of updating react-bootstrap.

File: client/webpack.rails.config.js (hunk 2 above)
Comment (justin808) [inline client/webpack.rails.config.js]: That's a valid point. I think the key is to not have babel parsing every one. Maybe it makes sense to create a separate test for that?

**Comment by justin808** [comment]
@bdefore Thanks for the feedback!
